src/lib/Repository/Values/Content/UiRelation.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository\Values\Content;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\Relation;
+use eZ\Publish\API\Repository\Values\Content\Relation as APIRelation;
+
+/**
+ * Extends original value object in order to provide additional fields.
+ */
+class UiRelation extends Relation
+{
+    /**
+     * Field definition name.
+     *
+     * @var string
+     */
+    protected $fieldDefinitionName;
+
+    /**
+     * The content type name for the destination.
+     *
+     * @var string
+     */
+    protected $destinationContentTypeName;
+
+    /**
+     * The location for the destination.
```

Comment:
This should be made more specific. The location is probably the main one. I'm also not sure about the phrasing. "Main location of the destination content" ?